duplex_http_call.go

```diff
@@ -169,7 +169,7 @@ func (d *duplexHTTPCall) CloseRead() error {
 
 // ResponseStatusCode is the response's HTTP status code.
 func (d *duplexHTTPCall) ResponseStatusCode() (int, error) {
-	<-d.responseReady
+	d.BlockUntilResponseReady()
```

Comment:
Noticed we were inconsistently using the `BlockUntilResponseReady` vs. directly accessing the channel.